airgun/views/ansible_variable.py

```diff
@@ -0,0 +1,129 @@
+        else:
+            raise "Missing the matcher key and values"
+
+    def read(self):
+        return {'matcher_key': self.matcher_key.read(), 'matcher_value': self.matcher_value.read()}
+
+
+class NewAnsibleVariableView(BaseLoggedInView):
+    """View while creating a new Ansible Variable"""
+
+    breadcrumb = BreadCrumb()
+    key = TextInput(id='ansible_variable_key')
+    description = TextInput(id='ansible_variable_description')
+    ansible_role = FilteredDropdown(id='ansible_variable_ansible_role_id')
+    override = Checkbox(id='ansible_variable_override')
+
+    # Accessing all widgets except the ones above requires that the `override` checkbox is filled
+    parameter_type = SatSelect(id='ansible_variable_parameter_type')
+    default_value = TextInput(id='ansible_variable_default_value')
+    hidden_value = Checkbox(id='ansible_variable_hidden_value')
+    expand_optional_input_validator = Text("//h2[@class='expander collapsed']")
+    required = Checkbox(id='ansible_variable_required')
+    validator_type = SatSelect(id='ansible_variable_validator_type')
+    validator_rule = TextInput(id='ansible_variable_validator_rule')
+    # the below attribute_order widget throws an error when a string value is passed to it
+    attribute_order = TextInput(id='order')
+    merge_overrides = Checkbox(id='ansible_variable_merge_overrides')
+    merge_default = Checkbox(id='ansible_variable_merge_default')
+    avoid_duplicates = Checkbox(id='ansible_variable_avoid_duplicates')
+    submit = Text('//input[@value="Submit"]')
+    cancel = Text("//a[contains(., text()='Cancel']")
+
+    @View.nested
+    class matcher_section(View):
+        add_matcher = Text("//a[contains(@class, 'add_nested_fields')]")
+        params = MatcherTable(
+            locator=".//table[@class='table white-header']",
+            column_widgets={
+                'Attribute type': MatcherActions(),
+                'Value': TextInput(locator=".//textarea[@id='new_lookup_value_value']"),
+                'Actions': Text(".//a"),
+            },
+        )
+
+        def before_fill(self, values):
+            if not self.params.is_displayed:
+                self.add_matcher.click()
+
+    @property
+    def expand_button(self):
+        """Return the Optional Input Validator section expander element"""
+        return self.browser.element(self.expand_optional_input_validator, parent=self)
+
+    @property
+    def expanded(self):
+        """Check whether this section is expanded"""
+        return 'active' in self.browser.get_attribute(
+            'class', self.expand_optional_input_validator
+        )
+
+    def expand(self):
+        """Expand the Optional Input Validator section"""
+        if not self.expanded:
+            self.browser.click(self.expand_optional_input_validator, parent=self)
+
+    @property
+    def is_displayed(self):
+        breadcrumb_loaded = self.browser.wait_for_element(self.breadcrumb, exception=False)
```

Comment:
Same consideration as the other class about using `wait_for` within is_displayed.

Reply:
Removed the `wait_for_element`.